front-end/src/app/components/achievements/achievements.component.ts

```diff
@@ -0,0 +1,41 @@
+import { ApiService } from './../../services/api.service';
+import { Component, OnInit } from '@angular/core';
+import { SettingsService } from 'src/app/services/settings.service';
+import { IUser } from '../user-profile/user-profile.component';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-achievements',
+  templateUrl: './achievements.component.html',
+  styleUrls: ['./achievements.component.scss']
+})
+export class AchievementsComponent implements OnInit {
+  public achievements: object;
+  private currentUser: IUser;
+  private subscriptions: Subscription[] = [];
+
+  constructor(
+    private apiService: ApiService,
+    settingsService: SettingsService,
+  ) {
+    this.run(); // work around for not being able to use async in constructor
```

Comment:
There is real no way around it - a constructor has to be synchronous.

However, you may wish to consider whether this should be in `ngOnInit` instead of the constructor. The constructor will only run once whereas ngOnInit will be triggered every time a component is shown (e.g. the page is changed).

You may only ever need to fix the achievements once though, since they don't change regularly.

Reply:
Agreed. I'll stick with it being in the constructor; as you said it's not like there will be new achievements every 5 minutes.